docs/config-options-listings.mdx

```diff
@@ -0,0 +1,61 @@
+---
+id: config-options-listings
+title: Listings For All BuildBuddy Options
+sidebar_label: Options Listings
+---
+
+import FreeServerConfig from '../bazel-bin/server/cmd/buildbuddy/yaml_doc/buildbuddy_server_documented_defaults.mdx'
+import EnterpriseServerConfig from '../bazel-bin/enterprise/server/cmd/server/yaml_doc/buildbuddy_enterprise_server_documented_defaults.mdx'
+import ExecutorConfig from '../bazel-bin/enterprise/server/cmd/executor/yaml_doc/buildbuddy_executor_documented_defaults.mdx'
+
+Provided below are working, documented YAML configs for each BuildBuddy binary
+containing every option that that binary accepts, each set to the default value
+for that option. Any option that can be specified in the YAML config can also be
+passed on the command line. For nested options, be sure to write out the full
+YAML path, with a `.` separating each part.
+
+For example:
+```yaml
+storage:
+    disk:
+        root_directory: /tmp/buildbuddy
+```
+becomes:
+```sh
+buildbuddy -storage.disk.root_directory="/tmp/buildbuddy"
+```
+
+For specifying lists of structures using flags on the command line, use the JSON
+representation of the list you wish to concatenate to the end or the element you
+wish to append:
+
+For example:
+```yaml
+cache:
+    disk:
+        partitions: [] # type: []disk.Partition
+        - id: "1GB"
+          max_size_bytes: 1073741824
+        - id: "2GB":
+          max_size_bytes: 2147483648
+```
+becomes:
```

Comment:
nit: it could be a bit more clear on how to determine whether something is specified as a struct or not. maybe something like:

---

For example, given the following schema:
```yaml
cache:
    disk:
        partitions: [] # type: []disk.Partition
        # e.g.:
        # - id: "" # type: string
        #   max_size_bytes: 0 # type: int
```

We see that `cache.disk.partitions` is configured as a list of `disk.Partition`. In YAML, we'd normally configure it like this:

```yaml
cache:
    disk:
        partitions:
            - id: "1GB"
              max_size_bytes: 1073741824
            - id: "2GB":
              max_size_bytes: 2147483648
```

The flag equivalent of this example would be:
...

Reply:
Done!